frontend/src/metabase/public/components/EmbedModal/StaticEmbedSetupPane/StaticEmbedSetupPane.tsx

```diff
@@ -174,48 +183,90 @@ export const StaticEmbedSetupPane = ({
 
       <Tabs defaultValue={TABS.Overview} data-testid="embedding-preview">
         <Tabs.List p="0 1.5rem">
-          <Tabs.Tab value={TABS.Overview}>{t`Overview`}</Tabs.Tab>
-          <Tabs.Tab value={TABS.Parameters}>{t`Parameters`}</Tabs.Tab>
-          <Tabs.Tab value={TABS.Appearance}>{t`Appearance`}</Tabs.Tab>
+          <Tabs.Tab
+            value={TABS.Overview}
+            onClick={() => setActiveTab(TABS.Overview)}
+          >{t`Overview`}</Tabs.Tab>
+          <Tabs.Tab
+            value={TABS.Parameters}
+            onClick={() => setActiveTab(TABS.Parameters)}
+          >{t`Parameters`}</Tabs.Tab>
+          <Tabs.Tab
+            value={TABS.Appearance}
+            onClick={() => setActiveTab(TABS.Appearance)}
+          >{t`Appearance`}</Tabs.Tab>
         </Tabs.List>
-        <Tabs.Panel value={TABS.Overview}>
-          <OverviewSettings
-            resourceType={resourceType}
-            selectedServerCodeOption={selectedServerCodeOption}
-            serverEmbedCodeSlot={getServerEmbedCodePane(TABS.Overview)}
-          />
-        </Tabs.Panel>
-        <Tabs.Panel value={TABS.Parameters}>
-          <ParametersSettings
-            activePane={activePane}
-            resourceType={resourceType}
-            resourceParameters={resourceParameters}
-            embeddingParams={embeddingParams}
-            lockedParameters={lockedParameters}
-            parameterValues={parameterValues}
-            iframeUrl={iframeUrl}
-            displayOptions={displayOptions}
-            serverEmbedCodeSlot={getServerEmbedCodePane(TABS.Parameters)}
-            onChangeEmbeddingParameters={setEmbeddingParams}
-            onChangeParameterValue={(id: string, value: string) =>
-              setParameterValues(state => ({
-                ...state,
-                [id]: value,
-              }))
-            }
-            onChangePane={setActivePane}
-          />
-        </Tabs.Panel>
-        <Tabs.Panel value={TABS.Appearance}>
-          <AppearanceSettings
-            activePane={activePane}
-            resourceType={resourceType}
-            iframeUrl={iframeUrl}
-            displayOptions={displayOptions}
-            serverEmbedCodeSlot={getServerEmbedCodePane(TABS.Appearance)}
-            onChangePane={setActivePane}
-            onChangeDisplayOptions={setDisplayOptions}
-          />
+        <Tabs.Panel value={activeTab}>
+          {activeTab === TABS.Overview ? (
+            <OverviewSettings
```

Comment:
I wish I could keep the preview state even when navigating to the Overview tab, but it's a bit hard to extract the content inside it here, so we only keep the embedding state when switching between Parameters and Appearance tabs.